### ISSUE TITLE
Services should be public for Symfony 4

### ISSUE THREAD

**Issue by Soullivaneuh** [liip/LiipFunctionalTestBundle#368]
To avoid this kind of error:

```
1) App\Tests\Controller\HookControllerTest::testNoAdminAccessWithRepositoryToken
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The "liip_functional_test.exception_listener" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead.

/code/vendor/symfony/dependency-injection/Container.php:252
/code/vendor/liip/functional-test-bundle/Utils/HttpAssertions.php:65
/code/vendor/liip/functional-test-bundle/Test/WebTestCase.php:915
/code/tests/Controller/HookControllerTest.php:198
```

**Comment by alexislefebvre** [comment]
This is a warning in Symfony 3.4 and it will stop working in Symfony 4, right?

**Comment by Soullivaneuh** [comment]
@alexislefebvre Exactly. More precisely, this king of error appears only if your tests are failing, but does not show the error of course.

**Comment by Soullivaneuh** [comment]
As a workaround waiting a fix, you can make it public from the kernel compiler pass:

```php
class Kernel extends BaseKernel implements CompilerPassInterface
{
    use MicroKernelTrait;

    public const CONFIG_EXTS = '.{php,xml,yaml,yml}';

    // Stuff...

    /**
     * {@inheritdoc}
     */
    public function process(ContainerBuilder $container)
    {
        $container->getDefinition('liip_functional_test.exception_listener')->setPublic(true);
    }
}
```

Ref: https://symfony.com/blog/new-in-symfony-3-4-defining-compiler-passes-in-the-kernel

**Comment by alexislefebvre** [comment]
It may have been fixed by #367.
